src/Groups/group_characters.jl

```diff
@@ -676,3 +736,79 @@ function character_field(chi::GAPGroupClassFunction)
 
     return F, MapFromFunc(f, finv, F, FF)
 end
+
+@doc Markdown.doc"""
+    schur_index(chi::GAPGroupClassFunction)
+
+Return either the minimal integer `m` such that the character `m * chi`
+is afforded by a representation over the character field of `chi`,
+or `nothing`.
+
+The latter happens if character theoretic criteria do not suffice for
+computing `m`.
+"""
+function schur_index(chi::GAPGroupClassFunction, recurse::Bool = true)
+    deg = numerator(degree(chi))
+    deg == 1 && return 1
+    indicator(chi) == -1 && return 2
+
+    # The character field contains an `m`-th root of unity.
+    values = chi.values
+    if conj(chi) == chi
+      bound = 2
+    else
+      # Compute the conductor of the largest cyclotomic field
+      # that is contained in the character field of `chi`.
+      gapfield = GAP.Globals.Field(values)
+      N = GAP.Globals.Conductor(gapfield)
+      for n in reverse(sort(divisors(N)))
+        if GAP.Globals.E(n) in gapfield
+          if isodd(n)
+            bound = 2*n
+          else
+            bound = n
+          end
+          break
+        end
+      end
+    end
+
+    # `m` divides `deg`
+    bound = gcd(bound, deg)
+    bound == 1 && return 1
+
+    # `m` divides the multiplicity of `chi` in any rational character
+    # with trivial Schur index.
+    # - Consider permutation characters induced from cyclic subgroups.
+    tbl = chi.table
+    for psi in induced_cyclic(tbl)
+      bound = gcd(bound, scalar_product(chi, psi))
+      bound == 1 && return 1
+    end
+    # - Consider characters induced from other known subgroups.
+    for name in names_of_fusion_sources(tbl)
+      s = character_table(name)
+      known, fus = known_class_fusion(s, tbl)
+      @assert known "the class fusion is not stored"
+      if length(class_positions_of_kernel(fus)) == 1
+        psi = trivial_character(s)^(tbl)
+        bound = gcd(bound, scalar_product(chi, psi))
+        bound == 1 && return 1
+      end
+    end
+
+    if recurse
+      # Consider tensor products of rational characters with Schur index 1.
+      cand = filter(psi -> degree(character_field(psi)[1]) == 1 &&
+                           schur_index(psi, false) == 1, collect(tbl))
+      for i in 1:length(cand)
+        for j in 1:i
+          bound = gcd(bound, scalar_product(chi, cand[i] * cand[j]))
+          bound == 1 && return 1
+        end
+      end
+    end
+
+    # For the moment, we do not have more character theoretic criteria.
+    return
```

Comment:
Matter of taste, but I'd usually prefer to be explicit about showing that we return `nothing` here, intentionally (as opposed to having forgotten to provide a return value). Feel free to ignore this!
```suggestion
    return nothing
```

Reply:
Fine.
(I had written `return` in order to make explicit that the return statement has not been forgotten.)